meteor/client/ui/Prompter/controller/joycon-device.ts

```diff
@@ -208,75 +201,94 @@ export class JoyConController extends ControllerAbstract {
 					typeof o.id === 'string' &&
 					o.id.match('STANDARD GAMEPAD Vendor: 057e')
 				) {
-					this.lastUsedJoyconIndex = o.index
-					this.lastUsedJoyconId = o.id
-					this.lastUsedJoyconMode =
+					const mode =
 						o.axes.length === 4
-							? 'LR'
+							? 'LR' // for documentation: L+R mode is also identified as Vendor: 057e Product: 200e
 							: o.id.match('Product: 2006')
 							? 'L'
 							: o.id.match('Product: 2007')
 							? 'R'
-							: null // we are setting lastUsedJoyconId as a member as opposed to returning it functional-style, to avoid doing this calculation pr. tick
-					// for documentation: L+R mode is also identified as Vendor: 057e Product: 200e
-					return { axes: o.axes, buttons: o.buttons }
+							: null
+					joyconInputs.push({ index: o.index, mode, axes: o.axes, buttons: o.buttons.map((i) => i.value) })
 				}
 			}
 		}
-		return false
+
+		return joyconInputs
 	}
 
-	private getActiveInputsOfJoycons(input) {
-		// handle buttons
-		// @todo: should this be throttled??
-		const newButtons = input.buttons.map((i) => i.value)
+	private getActiveInputsOfJoycons(joycons: JoyconWithData[]): number {
+		let lastSeenSpeed = 0
 
-		if (this.lastButtonArray.length) {
-			for (const i in newButtons) {
-				const oldBtn = this.lastButtonArray[i]
-				const newBtn = newButtons[i]
-				if (oldBtn === newBtn) continue
+		for (const joycon of joycons) {
+			// handle buttons at the same time as evaluating stick input
+			this.handleButtons(joycon)
 
-				if (!oldBtn && newBtn) {
-					// press
-					this.onButtonPressed(i, this.lastUsedJoyconMode)
-				} else if (oldBtn && !newBtn) {
-					// release
-					this.onButtonRelease(i, this.lastUsedJoyconMode)
+			// hadle speed input
+			if (joycon.mode === 'L' || joycon.mode === 'R') {
+				// L or R mode
+				if (Math.abs(joycon.axes[0]) > this.deadBand) {
+					if (joycon.mode === 'L') {
+						lastSeenSpeed = joycon.axes[0] * -1 // in this mode, L is "negative"
+					} else if (joycon.mode === 'R') {
+						lastSeenSpeed = joycon.axes[0] * 1.4 // in this mode, R is "positive"
+						// factor increased by 1.4 to account for the R joystick being less sensitive than L
+					}
 				}
-			}
-		}
-		this.lastButtonArray = newButtons
-
-		// hadle speed input
-		if (this.lastUsedJoyconMode === 'L' || this.lastUsedJoyconMode === 'R') {
-			// L or R mode
-			if (Math.abs(input.axes[0]) > this.deadBand) {
-				if (this.lastUsedJoyconMode === 'L') {
-					return input.axes[0] * -1 // in this mode, L is "negative"
-				} else if (this.lastUsedJoyconMode === 'R') {
-					return input.axes[0] * 1.4 // in this mode, R is "positive"
+			} else if (joycon.mode === 'LR') {
+				// L + R mode
+				// get the first one that is moving outside of the deadband, priorotizing the L controller
```

Comment:
Behaviour appears to be incorrect compared to this comment. Currently the right one will overwrite left if it moves out of the deadband.

Reply:
Fixed by adding the second clause in elseif instead of another if.